### PR TITLE
migrations: Deprecate use of _number of migrations_ in commands

### LINKED ISSUE #29544: non-sequential-migrations: Break num up/down convention
Currently our migration tools are controlled by a _number_ of migrations to perform up or down from the current version. Instead, what we should be doing is providing a *target* migration version to migrate to.

This should change the interface (and associated documentation) of the migrator and the sg migration command. This will have implementation changes in the migration definition package as well.

### PR DESCRIPTION
**Pull request by efritz**
Fixes #29544. This PR replaces the concept of migrating _n versions_ up or down with the concept of migrating _to_ a target version.

This changes the UI of `sg migration` as well as the migrator instance (which should currently have no explicit users that aren't default arguments).

### CODE REVIEW

**Comment by efritz** [comment]
- [ ] Update usage documentation

**Comment by LawnGnome** [comment]
I actually prefer the current behaviour, particularly for down, because it makes it easier to quickly test a migration you're in the middle of developing. (Although — in fairness — that's also more difficult with the `sg` workflow because of the way `io/fs` works.)

Could we keep "I just want to go down one version" somehow without needing to keep copy/pasting and editing the version number?

**Comment by efritz** [comment]
@LawnGnome Pushed a change to keep that behavior. WDYT?

**Review by LawnGnome — APPROVED** [review]
(Reviewing the down behaviour specifically.)

LGTM! I think that's more intuitive (and not particularly dangerous as a default, since it's only going down one level).

Thanks for making that change.

File: internal/database/migration/cliutil/flags.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func Down(commandName string, run RunFunc, out *output.Output) *ffcli.Command {
 	var (
 		downFlagSet          = flag.NewFlagSet(fmt.Sprintf("%s down", commandName), flag.ExitOnError)
 		downDatabaseNameFlag = downFlagSet.String("db", "", "The target database instance.")
-		downNFlag            = downFlagSet.Int("n", 1, "How many migrations to apply.")
+		downTargetFlag       = downFlagSet.Int("target", 0, "Reset all migrations defined after this target. Zero (the default) reverts the latest migration.")
```
Comment (unknwon) [inline internal/database/migration/cliutil/flags.go:82]: qq: If I keep giving zeros to migration down, I should be able to keep reverting migrations right? (i.e. 3 -> 2 -> 1)

**Review by mrnugget — APPROVED** [review]

**Review by eseliger — APPROVED** [review]

File: internal/database/migration/cliutil/flags.go (hunk 1 above)
Comment (efritz) [reply internal/database/migration/cliutil/flags.go:82]: For now (that we still have sequentially orderable migrations), yes.

Comment (unknwon) [reply internal/database/migration/cliutil/flags.go:82]: >For now (that we still have sequentially orderable migrations), yes.

Sounds like for later this is a "no"?

**Pull request merged**